### PR TITLE
Feature #9108 implemented. A scroll to top anchor tag added.

### PR DESCRIPTION
**Pull request by nishantjo-c**
Fixes: #9108  

Smooth scroll back to the top implemented for user convenience.

### CODE REVIEW

**Review by Falke-Design — CHANGES_REQUESTED** [review]
This is a good start but there are a few things missing. Please checkout the must-have list: https://github.com/Leaflet/Leaflet/issues/9108#issuecomment-1742200671

And please make sure to only commit the relevant changes and don't change any formatting:

- Revert Gemfile.lock
- Revert blanks in v2.html
- Revert main.css
- Revert formatting in docs.js

**Comment by nishantjo-c** [comment]
I have added the screenshots in the previous commit. 
The design is changes as required.

File: docs/Gemfile.lock [hunk 1]
```diff
@@ -250,6 +252,7 @@ GEM
     webrick (1.8.1)
 
 PLATFORMS
+  x64-mingw-ucrt
```
Comment (Falke-Design) [inline docs/Gemfile.lock:255]:
```suggestion
```

File: docs/_layouts/v2.html [hunk 2]
```diff
@@ -57,6 +57,8 @@
 	</script>
 </head>
 <body{% if page.bodyclass %} class="{{ page.bodyclass }}"{% endif %}>
+	<!-- back to top -->
+<!-- <div id="backtotop"><div class="backtotop"></div></div> -->
```
Comment (Falke-Design) [inline docs/_layouts/v2.html:61]:
```suggestion
```

File: docs/_layouts/v2.html [hunk 3]
```diff
@@ -120,12 +122,14 @@ <h1><a href="https://leafletjs.com/"><img src="{{ root }}docs/images/logo.png" a
...
-
 	<nav class="ext-links">
-	  <a class="ext-link twitter" href="https://twitter.com/LeafletJS" title="Follow LeafletJS on Twitter"><img alt="Follow LeafletJS on Twitter" src="{{root}}docs/images/twitter-round.svg" width="46" /></a>
+		<a class="ext-link twitter" href="https://twitter.com/LeafletJS" title="Follow LeafletJS on Twitter"><img alt="Follow LeafletJS on Twitter" src="{{root}}docs/images/twitter-round.svg" width="46" /></a>
 	  <a class="ext-link github" href="http://github.com/Leaflet/Leaflet" title="View Source on GitHub"><img alt="View Source on GitHub" src="{{root}}docs/images/github-round.svg" width="46" /></a>
 	  <a class="ext-link forum" href="https://stackoverflow.com/questions/tagged/leaflet" title="Ask for help on Stack Overflow"><img alt="Leaflet questions on Stack Overflow" src="{{root}}docs/images/forum-round.svg" width="46" /></a>
+		<div id="backtotop" class="ext-link"><div class="backtotop"></div></div>
 	</nav>
+
+
```
Comment (Falke-Design) [inline docs/_layouts/v2.html:132]:
```suggestion
```

File: docs/docs/js/docs.js [hunk 4]
```diff
@@ -56,7 +56,7 @@ if (toc) {
 	const scrollPos = function scrollPos() {
 		const scroll = window.scrollY;
 
-		if (scroll >= (toc.offsetHeight + toc.offsetTop)) {
+		if (scroll >= toc.offsetHeight + toc.offsetTop) {
```
Comment (Falke-Design) [inline docs/docs/js/docs.js:59]: Don't format code which you don't change
```suggestion
		if (scroll >= (toc.offsetHeight + toc.offsetTop)) {
```

**Review by Falke-Design — CHANGES_REQUESTED** [review]
Please place the "back-to-top" button on the right bottom corner and make it much smaller and also make the arrow centered

File: docs/docs/js/docs.js (hunk 4 above)
Comment (nishantjo-c) [reply docs/docs/js/docs.js:59]: Hi, I am using prettier in my code for auto formatting, I think that would be auto formatting the code itself. Sorry for the inconvenience, I will disable it before working further.

Comment (nishantjo-c) [reply docs/docs/js/docs.js:59]: Hi, I am using prettier in my code for auto formatting, I think that would be auto formatting the code itself. Sorry for the inconvenience, I will disable it before working further.

Comment (nishantjo-c) [reply docs/docs/js/docs.js:59]: Also, in the desktop mode, would you prefer having the scroll button on the right bottom as well or just in the tab and mobile mode?

Comment (Falke-Design) [reply docs/docs/js/docs.js:59]: on all devices on the right bottom

**Comment by nishantjo-c** [comment]
Changes made. Made the button response with screen size. 
Review the design and tell me if some more changes required.

File: docs/docs/css/main.css [hunk 5]
```diff
@@ -147,6 +147,35 @@ h4 {
...
+	height: 36px;
+	width: 36px;
+	background-color: #199900;
+	border-radius: 50%;
+	transition: all .2s;
+	z-index: 400;
+	
+	&:hover{
+		opacity: .5;
+	}
```
Comment (Falke-Design) [inline docs/docs/css/main.css:166]: Please keep the classic css definition for now.

```
#backtotop:hover{
		opacity: .5;
	}
```

File: docs/docs/css/main.css [hunk 6]
```diff
@@ -1015,6 +1056,18 @@ tr:target {
 		line-height: 1;
 	}
 
+	#backtotop{
+		width: 26px;
+		height: 26px;
+		right: .2em;
+	}
+	.backtotop{
```
Comment (Falke-Design) [inline docs/docs/css/main.css:1064]: Please rename the class or the id. They should not have the same name

File: docs/_layouts/v2.html [hunk 7]
```diff
@@ -120,12 +122,14 @@ <h1><a href="https://leafletjs.com/"><img src="{{ root }}docs/images/logo.png" a
 	<div class="footer">
 		<p>&copy; 2010–{{ 'now' | date: "%Y" }} <a href="https://agafonkin.com">Volodymyr Agafonkin</a>. Maps &copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors.</p>
 	</div>
-
 	<nav class="ext-links">
-	  <a class="ext-link twitter" href="https://twitter.com/LeafletJS" title="Follow LeafletJS on Twitter"><img alt="Follow LeafletJS on Twitter" src="{{root}}docs/images/twitter-round.svg" width="46" /></a>
+		<a class="ext-link twitter" href="https://twitter.com/LeafletJS" title="Follow LeafletJS on Twitter"><img alt="Follow LeafletJS on Twitter" src="{{root}}docs/images/twitter-round.svg" width="46" /></a>
 	  <a class="ext-link github" href="http://github.com/Leaflet/Leaflet" title="View Source on GitHub"><img alt="View Source on GitHub" src="{{root}}docs/images/github-round.svg" width="46" /></a>
 	  <a class="ext-link forum" href="https://stackoverflow.com/questions/tagged/leaflet" title="Ask for help on Stack Overflow"><img alt="Leaflet questions on Stack Overflow" src="{{root}}docs/images/forum-round.svg" width="46" /></a>
+		<div id="backtotop" class="ext-link"><div class="backtotop"></div></div>
```
Comment (Falke-Design) [inline docs/_layouts/v2.html:129]: I think it makes sense, to have this not in this `<nav>` element, because it is displayed at the bottom

**Review by Falke-Design — CHANGES_REQUESTED** [review]
Please revert the not necessary changes like in the `Gemfile` and the comments in `v2.html`. I will not review your code again until this changes are reverted.

Also please update your branch with the newest main branch

File: docs/_layouts/v2.html (hunk 7 above)
Comment (nishantjo-c) [reply docs/_layouts/v2.html:129]: Yeah! Before it was just next to the nav elements. And then I re positioned it to the right bottom. I will re-position it to somewhere else.

**Comment by nishantjo-c** [comment]
Hey i have made changes and requested a review.

**Comment by Falke-Design** [comment]
> Please **revert** the not necessary changes like in the **Gemfile** and the comments in **v2.html**. I will not review your code again until this changes are reverted.
Also please **update your branch** with the newest main branch

**Comment by Falke-Design** [comment]
Superseded by #9186

**Pull request closed without merging**